source/acquaman/AMGenericContinuousScanController.cpp

```diff
@@ -0,0 +1,143 @@
+#include "beamline/AMBeamline.h"
+
+AMGenericContinuousScanController::AMGenericContinuousScanController(AMGenericContinuousScanConfiguration *configuration, QObject *parent)
+	: AMContinuousScanActionController(configuration, parent)
+{
+	configuration_ = configuration;
+
+	scan_ = new AMScan();
+	scan_->setName(configuration_->name());
+	scan_->setScanConfiguration(configuration_);
+	scan_->setFileFormat("amCDFv1");
+	scan_->setIndexType("fileSystem");
+
+	AMControlInfo axisControlInfo1 = configuration_->axisControlInfos().at(0);
+	scan_->rawData()->addScanAxis(AMAxisInfo(axisControlInfo1.name(), 0, axisControlInfo1.description()));
+
+	if (configuration_->scanAxes().size() == 2){
+
+		AMControlInfo axisControlInfo2 = configuration_->axisControlInfos().at(1);
+		scan_->rawData()->addScanAxis(AMAxisInfo(axisControlInfo2.name(), configuration_->scanAxisAt(1)->numberOfPoints(), axisControlInfo2.description()));
+	}
+
+	secondsElapsed_ = 0;
+	secondsTotal_ = configuration_->totalTime();
+	elapsedTime_.setInterval(1000);
+	connect(this, SIGNAL(started()), &elapsedTime_, SLOT(start()));
+	connect(this, SIGNAL(cancelled()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(paused()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(resumed()), &elapsedTime_, SLOT(start()));
+	connect(this, SIGNAL(failed()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(finished()), &elapsedTime_, SLOT(stop()));
+	connect(&elapsedTime_, SIGNAL(timeout()), this, SLOT(onScanTimerUpdate()));
+}
+
+AMGenericContinuousScanController::~AMGenericContinuousScanController()
+{
+
+}
+
+void AMGenericContinuousScanController::onScanTimerUpdate()
+{
+	if (elapsedTime_.isActive()){
+
+		if (secondsElapsed_ >= secondsTotal_)
+			secondsElapsed_ = secondsTotal_;
+		else
+			secondsElapsed_ += 1.0;
+
+		emit progress(secondsElapsed_, secondsTotal_);
+	}
+}
+
+AMAction3 *AMGenericContinuousScanController::createInitializationActions()
+{
+	return 0;
+}
+
+AMAction3 *AMGenericContinuousScanController::createCleanupActions()
+{
+	return 0;
+}
+
+void AMGenericContinuousScanController::buildScanControllerImplementation()
+{
+	QList<AMXRFDetector *> xrfDetectors;
+	QList<AMDataSource *> spectrumSources;
+
+	for (int i = 0, size = configuration_->detectorConfigurations().count(); i < size; i++){
+
+		AMXRFDetector *xrfDetector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByInfo(configuration_->detectorConfigurations().at(i)));
+
+		if (xrfDetector){
+
+			xrfDetectors << xrfDetector;
+			spectrumSources << scan_->dataSourceAt(scan_->indexOfDataSource(xrfDetector->name()));
+		}
+	}
+
+	if (!xrfDetectors.isEmpty()){
+
+		// This is sufficient to clear all regions of interest on all detectors as they should by synchronized via AMBeamline::synchronizeXRFDetectors.
+		AMXRFDetector *detector = xrfDetectors.first();
+		detector->removeAllRegionsOfInterest();
+
+		AMDataSource *spectrumSource = 0;
+
+//		if (xrfDetectors.size() > 1){
```

Comment:
@dretrex You mentioned that you might have had a fix in place for the issue that caused this code to be commented out. Is that still true?